lib/date.nit

```diff
@@ -0,0 +1,215 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# The date and time behavior is splited in 2 classes:
+# * `Date` handles the year, month and day part of the date.
+# * `Time` handles the time in hour, minute and second.
+# 
+# These are extended by DateTime that merges the two concepts in one class.
+#
+#	###################
+#	# Some operations #
+#	###################
+#
+#	#DateTime:
+#	var date_1  = new DateTime(2014, 02, 09, 13, 50, 0)
+#	var date_2  = new DateTime(2014, 02, 09, 13, 51, 0)
+#	
+#	assert date_1 > date_2
+# 	
+#	#Date:
+#	var date_1 = new Date(2014, 01, 12)
+#	var date_2 = new Date(2014, 01, 30)
+#	
+#	assert date_1.diff_month(date_2)
+#
+#	#Time:
+#	var time_1 = new Time.now
+#	var time_2 = new Time(14,25,30)
+#	
+#	assert time_1 == time_2
+#
+module date
+
+# The class "Time" allow the manipulation only on the time data
+#
+# 		var now = new Time.now ==> Set the current time
+# 		var time = new Time(14,30,00) ==> (Hours, Minutes, Seconds)
+class Time
+	super Comparable
+
+	# Hours as an integer
+	var hour: Int
+
+	# Minutes as an integer
+	var minute: Int
+
+	# Seconds as an integer
+	var second: Int
+
+	# This will initialize the current time
+	init now do
+		var tm = new Tm.localtime
+		hour = tm.hour
+		minute = tm.min
+		second = tm.sec
+	end
+
+	redef fun ==(d)  
+	do
+		return d isa Time and time_eq(d)
+	end
+
+	# This check if the time is eq or not
+	fun time_eq(d: Time): Bool
+	do
+		if (hour * 3600 + minute * 60 + second) != (d.hour * 3600 + d.minute * 60 + d.second) then
```

Comment:
You should add a `to_milisec` ou `timestamp` method to `Date` to make that.

The result should looks like:

~~~nit
return timestamp == d.timestamp
~~~